pkg/webservice/handlers.go

```diff
@@ -642,6 +642,53 @@ func getPartitionQueues(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPartitionQueue(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+	vars := httprouter.ParamsFromContext(r.Context())
+	if vars == nil {
+		buildJSONErrorResponse(w, MissingParamsName, http.StatusBadRequest)
+		return
+	}
+	partition := vars.ByName("partition")
+	partitionContext := schedulerContext.GetPartitionWithoutClusterID(partition)
+	if partitionContext == nil {
+		buildJSONErrorResponse(w, PartitionDoesNotExists, http.StatusNotFound)
+		return
+	}
+	queueName := vars.ByName("queue")
+	queueErr := validateQueue(queueName)
+	if queueErr != nil {
+		buildJSONErrorResponse(w, queueErr.Error(), http.StatusBadRequest)
+		return
+	}
+	queue := partitionContext.GetQueue(queueName)
+	if queue == nil {
+		buildJSONErrorResponse(w, QueueDoesNotExists, http.StatusNotFound)
+		return
+	}
+	// The default is to exclude children
+	queueDao := queue.GetPartitionQueueDAOInfo(true)
+	// exclude children
+	if exclude := strings.ToLower(r.URL.Query().Get("exclude")); exclude != "" {
+		if exclude != "children" {
+			buildJSONErrorResponse(w, "Only following exclude is allowed: children", http.StatusBadRequest)
+			return
+		}
+	}
+	// include children
+	if include := strings.ToLower(r.URL.Query().Get("include")); include != "" {
+		if include == "children" {
+			queueDao = queue.GetPartitionQueueDAOInfo(false)
+		} else {
+			buildJSONErrorResponse(w, "Only following include is allowed: children", http.StatusBadRequest)
+			return
+		}
+	}
```

Comment:
A simple `subtree` query parameter is a much more elegant solution.
If the parameter is there we call `GetPartitionQueueDAOInfo(true)` otherwise `GetPartitionQueueDAOInfo(false)` The default, and not present, is `false` so code wise it would look a bit like this:
```
queueDao := queue.GetPartitionQueueDAOInfo(r.URL.Query().Has("subtree"))
```
It checks for the presence of the key `subtree`